### PR TITLE
Make the stage color picker work with touch.

### PR DESCRIPTION
**Pull request by paulkaplan**
### Resolves

_What Github issue does this resolve (please include link)?_

- Resolves https://github.com/LLK/scratch-gui/issues/1350

### Proposed Changes

_Describe what this Pull Request does_

Make the color picker on the stage work the same way the costume editor one does: show the picker on _either_  mouse down or mouse move, then submit the color on mouse up. This allows it to work on touch, where you can touch down to see the loupe and drag around to pick your color.

### Reason for Changes

_Explain why these changes should be made_

The touching color block couldn't be used on touch!

----

A lot of this change is just moving blocks of code around.

This PR can be tested on touch devices at https://llk.github.io/scratch-gui/touch-picker/

### CODE REVIEW

**Comment by rschamp** [comment]
Try it out: [touch-picker](https://llk.github.io/scratch-gui/touch-picker)

_I'm a bot, not actually @rschamp!_

**Comment by paulkaplan** [comment]
@rschamp thanks for catching that. Fixed.

**Comment by paulkaplan** [comment]
I think we should file an issue for the growing complexity of `stage.jsx`. I don't like that modifying the logic for the color picker behavior is right next to modifying the drag behavior

File: src/containers/stage.jsx [hunk 1]
```diff
@@ -112,7 +112,9 @@ class Stage extends React.Component {
     }
     startColorPickingLoop () {
         this.intervalId = setInterval(() => {
-            this.setState({colorInfo: this.getColorInfo(this.pickX, this.pickY)});
+            if (typeof this.pickX === 'number') {
```
Comment (rschamp) [inline src/containers/stage.jsx]: What situation is this not true, before it's initialized?

**Pull request merged**